frontend/user/profileService.js

```diff
@@ -29,5 +44,55 @@
             return HttpService.patch(USER_URI, data);
         };
 
-    });
+        service._isAdmin = institution => user.isAdmin(institution.key);
+
+        service.removeProfile = (event, institution) => {
+            if (!service._isAdmin(institution)) {
+                const textContent = service._hasMoreThanOneInstitution() ? HAS_MORE_THAN_ONE_INSTITUTION_MSG : HAS_ONLY_ONE_INSTITUTION_MSG;
+                const confirm = $mdDialog.confirm();
+                confirm
+                    .clickOutsideToClose(false)
+                    .title('Remover vínculo com ' + institution.name)
+                    .textContent(textContent)
+                    .ariaLabel('Remover instituicao')
+                    .targetEvent(event)
+                    .ok('Sim')
+                    .cancel('Não');
+                const promise = $mdDialog.show(confirm);
+                promise.then(function () {
+                    service._deleteInstitution(institution.key);
+                }, function () {
+                    MessageService.showToast('Cancelado');
+                });
+                return promise;
+            } else {
+                MessageService.showToast('Desvínculo não permitido. Você é administrador dessa instituição.');
+            }
+        };
+
+        service._deleteInstitution = (institutionKey) => {
+            return new Promise(resolve => {
+                UserService.deleteInstitution(institutionKey)
+                    .then(_ => {
+                        service._removeConnection(institutionKey);
+                        resolve();
+                    });
+            });
+        };
+
+        service._removeConnection = (institutionKey) => {
+            if (service._hasMoreThanOneInstitution()) {
+                user.removeInstitution(institutionKey);
```

Comment:
why remove institution and profile?

Reply:
The institution must be removed from the user's institution because he is no longer a member, as well his respective profile

Reply:
ok, but why it only happens when the user has more than one institution?